contents/blog/decouple-deployment-from-release.md

```diff
@@ -0,0 +1,113 @@
+
+## What are the differences between deployment and release?
+
+The process of delivering code starts with writing and testing code locally. After completing a piece of code, it often goes through reviews and more tests before integrating with existing code. 
+
+Once integrated, the new code deploys. Deployment means getting code from a local machine or repository into production. Your application or build containing the new code becomes accessible on production infrastructure.
+
+After deploying successfully, the code releases for users to start using it. The application exposes in production and the functionality becomes available. A release comes after a deployment, and by default, couples to it. This means it happens immediately after a deployment succeeds.
+
+![Delivery process](../images/blog/decouple-deployment-from-release/delivery.png)
+
+This whole process is the software delivery process. Deployment and release are two separate, but critical parts of it.
+
+## Why should you decouple deployment and release?
+
+Without restrictions, deployed code is released code. This coupling of deployment and release can cause two main issues:
+
+1. Problems isolated to the new code due to deployment. For example, a value saving locally but not in production due to an unexpected interaction with the production database.
+
+2. Problems with the entire release due to new code. This is a broken release. For example, values created by new features corrupt other data unexpectedly in the production database.
+
+Decoupling deployment from release lowers the risk of these issues. This is because it enables testing code in production before releasing it. You can identify issues arising from production infrastructure and integration with the final, real release. 
+
+On top of lowering the risk of issues, decoupling can increase developer productivity and app reliability. The specific benefits include:
+
+- Since developers don’t need to wait for a release to deploy, they can ship, [test in production](/blog/testing-in-production), and get feedback faster.
+
+- By helping developers worry less about breaking the release, decoupling creates psychological safety. It makes developers more productive knowing they can roll back a change easily without impacting users.
+
+- Developers can pass off releases to product managers and marketers once they deploy. They can continue to code instead of needing to coordinate with stakeholders.
+
+- Building the infrastructure for rollbacks (like [feature flags](/feature-flags)) makes rollbacks easier and decreases user impact. Creates less downtime when there is an issue.
+
+- Promote trunk-based development by merging to `main` more often instead of changes waiting on branches. Reduces merge conflicts.
```

Comment:
What is trunk-based development? (I know what it is, but Im not sure everyone will. Maybe lets hyperlink to more info on this?)
```suggestion
- Promote trunk-based development by merging to `main` more often instead of changes waiting on branches. Reduces merge conflicts.
```

Reply:
Clarified